common/src/main/java/com/wynntils/core/functions/templates/ExpressionTemplatePart.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.functions.templates;
+
+import com.wynntils.core.functions.expressions.Expression;
+import com.wynntils.core.functions.expressions.parser.ExpressionParser;
+import com.wynntils.utils.type.ErrorOr;
+
+public class ExpressionTemplatePart extends TemplatePart {
+    private final String expressionString;
+
+    public ExpressionTemplatePart(String part) {
+        super(part);
+
+        if (!this.part.startsWith("{") || !this.part.endsWith("}")) {
+            throw new IllegalArgumentException("Expression was not wrapped in curly braces.");
```

Comment:
I'm getting a bit lost in the data flow here, sorry. If the user enters a format string like `unmatched: {mana_max`, will we end up here then? Or is this supposed to be caught before by the parser, so this is just a sanity check? And if we end up here, how does this exception propagate to something nicely for the user to see?

Reply:
This is more of a "there is no way this can happen at the moment since the callers are safe but I left this here".

It is fine to remove. But it is also fine to keep

Reply:
So, what do you think @magicus ?

Reply:
That was basically what I wanted to hear, that is just a fail-safe. Thanks.